src/picongpu/include/fields/background/templates/bgrTWTS.hpp

```diff
@@ -0,0 +1,953 @@
+             
+            return eFieldPositions_SI;
+        }
+         
+        template<>
+        HDINLINE PMacc::math::Vector<float3_64,FieldE::numComponents>
+        TWTSFieldE::getEfieldPositions_SI<DIM2>(const DataSpace<simDim>& cellIdx) const
+        {
+            const float2_64 cellDimensions = ( float_64( picongpu::cellSize.x() ),
+                                               float_64( picongpu::cellSize.y() ) )
+                                             * unit_length;
+            
+            /* TWTS laser coordinate origin is centered transversally and defined longitudinally by
+               the laser center in y (usually maximum of intensity). */
+            float2_X laserOrigin = float2_X( halfSimSize.x(), focus_y_SI/cellDimensions.y() );
+            
+            /* For the Yee-Cell shifted fields, obtain the fractional cell index components and add
+             * that to the total cell indices. The physical field coordinate origin is transversally
+             * centered with respect to the global simulation volume. */
+            PMacc::math::Vector<float3_X, FieldE::numComponents> eFieldPositions =
+                                    dummyCast(fieldSolver::NumericalCellType::getEFieldPosition());
+            
+            PMacc::math::Vector<float3_64,FieldE::numComponents> eFieldPositions_SI;
+            
+            for( uint32_t i = 0; i < FieldE::numComponents; ++i ) /* cellIdx Ex, Ey and Ez */
+            {
+                eFieldPositions[i]+=float3_X( float_X(cellIdx.x() - laserOrigin.x()),
+                                              float_X(cellIdx.y() - laserOrigin.y()),
+                                              float_X(0.0) );
+                
+                eFieldPositions_SI[i]=float3_64( float_64( (eFieldPositions[i]).x() )
+                                                                               * cellDimensions.x(),
+                                                 float_64( (eFieldPositions[i]).y() )
+                                                                               * cellDimensions.y(),
+                                                 float_64(0.0) );
+                
+                /*  Rotate 90 degree around y-axis, so that TWTS laser propagates within
+                 *  the 2D (x,y)-plane. Corresponding position vector for the Ez-components
+                 *  in 2D simulations.
+                 *  3D     3D vectors in 2D space (x,y)
+                 *  x -->  z
+                 *  y -->  y
+                 *  z --> -x (Since z=0 for 2D, we use the existing
+                 *            TWTS-field-function and set -x=0)
+                 *  Ex --> Ez (--> Same function values can be used in 2D,
+                 *             but with Yee-Cell-Positions for Ez.)
+                 *  By --> By
+                 *  Bz --> -Bx
+                 */
+                eFieldPositions_SI[i] = float3_64( -(eFieldPositions_SI[i]).z(),
```

Comment:
`TwistVector` shuffles, but does not include signs. I discussed this with @Heikman . Furthermore I would argue that `TwistVector` does not greatly improve readability in this specific case. Hence this would be just an other dependency with no gain besides syntactic sugar.